File: 0000_0100/0009_palindrome_number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        string num = to_string(x);
        int len = num.length();
        int i, max = len / 2;
        
        for (i = 0; i < max && num[i] == num[len - i - 1]; i++);
        
        return i == max;
    }
};

File: 0000_01000/0009_palindrome_number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        
        if x < 0 {
            return false
        }
        let (mut tmp, mut rev): (i32, i32) = (x, 0);

        while tmp != 0 {
            rev = rev * 10 + tmp % 10;
            tmp /=10;
        }
        x == rev
    }
}
